internal/ast/types.go

```diff
@@ -666,6 +667,33 @@ type EnumType struct {
 	Values []EnumValue // possible values. Value types might be different
 }
 
+func (t EnumType) MemberForValue(value any) (EnumValue, bool) {
+	if len(t.Values) == 0 {
+		return EnumValue{}, false
+	}
+
+	if value == nil {
+		return t.Values[0], false
+	}
+
+	equal := func(a, b any) bool {
```

Comment:
What is the benefit of that function?

Reply:
To work around a kink caused by using `any` in our IR. Because of that, some numerical values are stored as an `int` (within an `any` field) while others can be `int64`, `int32`, ...
Which means that for now, in order to have equality checks that work as expected, we need to do some weird things :/